### PR TITLE
feat(Dropdown): Add custom item support examples

### LINKED ISSUE #1163: [Dropdown] Add remaining examples
The [web did a lot of examples](https://sparkui.vercel.app/?path=/docs/components-dropdown--docs#multiple-selection) for this component, let's import them in the catalog app.

### PR DESCRIPTION
**Pull request by soulcramer**
## 📋 Changes

Add Custom items examples and improve the configurator to actually represent the selection when an item is tapped.
Also fix the Dropdown showing the menu while the Textfield is not enabled.

## 🤔 Context

Close #1163 

## ✅ Checklist

- [x] I have reviewed the submitted code.
- [x] I have tested on a phone device/emulator.
- [x] If it includes design changes, please ask for a review `spark-design` GitHub team.

## 🗒️ Other info

A11y need some fixes in the examples

## Summary by CodeRabbit

- **New Features**
  - Introduced new `DropdownMenuGroupItem` component for dropdown groups.
  - Added new custom dropdown item examples (Single/Multi-Select, Custom Items).

- **Enhancements**
  - Improved text element styling with new highlight style.
  - Updated dropdown text field state message styling.
  - Refactored dropdown examples into separate composable functions.

- **Bug Fixes**
  - Prevented menu expansion when the text field is disabled.
  - Adjusted progress indicator mappings for better Spark compatibility.

- **Refactor**
  - Reformatted class declarations for better readability.
  - Removed unnecessary imports and reorganized component structure.

### CODE REVIEW

**Review by SimonMarquis — CHANGES_REQUESTED** [review]
Can you only submit the actual code change related to this PR, and create another PR containing all the other formatting changes?

**Comment by soulcramer** [comment]
I don't know what happened here 🤔 , I only rebased from main and run spotless

**Comment by kazaky** [comment]
@coderabbitai review

**Comment by kazaky** [comment]
@coderabbitai summary

File: catalog/src/main/kotlin/com/adevinta/spark/catalog/configurator/samples/textfields/DropdownsConfigurator.kt [hunk 1]
```diff
@@ -102,14 +107,34 @@ private fun ColumnScope.DropdownSample() {
...
+                title = {
+                    Text(groupName)
+                },
+            ) {
+                books.forEach { book ->
+                    DropdownMenuItem(
+                        text = { Text(book) },
+                        onClick = {
+                            valueText = book
+                            if(hideOnSelect) expanded = false
```
Comment (SimonMarquis) [inline catalog/src/main/kotlin/com/adevinta/spark/catalog/configurator/samples/textfields/DropdownsConfigurator.kt:121]: You'll probably have to re-run spotless or manually reformat the files
```suggestion
                            if (hideOnSelect) expanded = false
```

**Comment by soulcramer** [comment]
@spark-ui-bot spotless

**Comment by kazaky** [comment]
@spark-ui-bot paparazzi

**Comment by soulcramer** [comment]
@spark-ui-bot spotless

**Pull request merged**